modules/passkey/src/utils/webauthn.ts

```diff
@@ -0,0 +1,185 @@
+import { ethers } from 'ethers'
+import type { BigNumberish, BytesLike } from 'ethers'
+import CBOR from 'cbor'
```

Comment:
This isn't great because now the package depends on `ethers` which ideally would be great to avoid. Is there a way to remove this dependency; i.e. can we implement some helper functions to replace the `ethers` functionality we use?

If my memory serves me well, we only use it for byte manipulation which doesn't merit pulling in the entire `ethers` dependency for.

`cobr` makes sense though.

Reply:
`ethers` is used for:
- `getBytes`
- `concat`
- `sha256`
- `toUtf8Bytes`
- `isBytesLike`
- `encodeBase64`
- `hexlify`
- `MaxUint256`
- `AbiCoder`
- `solidityPacked`
- `toBeHex`

Reply:
Maybe this will be helpful, but in the example app I implemented public key extraction without the `cbor`, using only built-in `crypto` module: https://github.com/safe-global/safe-modules/blob/89c46b2c9a27fec6ffbaefc6391742e6b76e41c3/examples/4337-passkeys/src/logic/passkeys.ts#L57-L68

(This would require double-checking that the module exists in the node.js)

Reply:
Appears to be the case: https://nodejs.org/docs/latest/api/crypto.html#cryptosubtle and https://nodejs.org/docs/latest/api/webcrypto.html#class-subtlecrypto

Reply:
- `getBytes`: not needed if we use `Uint8Array` everywhere
- `concat`: easy to implement
- `sha256`: shouldn't be needed, added comment :point_down: 
- `toUtf8Bytes`: Trivial with `TextEncoder` API
- `isBytesLike`: shouldn't be needed if base-64 URL encoding is implemented without `ethers`
- `encodeBase64`: shouldn't be hard, there are lots of implementations floating around the internet.
- `hexlify`: should be trivial to implement
- `MaxUint256`: trivial
- `AbiCoder`: Should be implemented in function of concatenation, we don't need general ABI encoding support
- `solidityPacked`: idem
- `toBeHex`: should be easy to implement.

I also have a feeling that many of these methods will no longer be needed once you start de-ethers-ifying things.

Reply:
Done 👍🏾